### PR TITLE
[RPA-345] Action Rework: Api request

### PR DESCRIPTION
**Pull request by LukaszKaczkowski**

### CODE REVIEW

File: runbotics/runbotics-desktop/src/config/BackgroundPageApiRequestHandler.ts [hunk 1]
```diff
@@ -56,21 +56,34 @@ export class BackgroundPageApiRequestHandler extends StatelessActionHandler impl
...
+        const method = input.method ? input.method : 'GET';
         try {
             switch (method) {
-                case 'get':
+                case 'GET':
                     response = await Axios.get(compiled, { headers: input.headers });
                     break;
-                case 'post':
+                case 'DELETE':
+                    response = await Axios.delete(compiled, JSON.parse(input.body));
```
Comment (FranekMarciniak) [inline runbotics/runbotics-desktop/src/config/BackgroundPageApiRequestHandler.ts]: Delete does not need `input.body`

Comment (FranekMarciniak) [reply runbotics/runbotics-desktop/src/config/BackgroundPageApiRequestHandler.ts]: Look at GET method

Comment (FranekMarciniak) [reply runbotics/runbotics-desktop/src/config/BackgroundPageApiRequestHandler.ts]: And add `input.headers`

File: runbotics/runbotics-desktop/src/config/BackgroundPageApiRequestHandler.ts [hunk 2]
```diff
@@ -121,7 +134,7 @@ export type ApiRequestInput = ApiResource & {
 };
 
 export type ApiRequestOutput<T> = {
-    data: T;
+    data?: T;
```
Comment (FranekMarciniak) [inline runbotics/runbotics-desktop/src/config/BackgroundPageApiRequestHandler.ts]: We can to previous version where data is required

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/Actions/api.actions.ts [hunk 3]
```diff
@@ -46,14 +46,14 @@ const getApiActions: () => Record<string, IBpmnAction> = () => ({
                                     {
                                         properties: {
                                             method: {
-                                                enum: ['get'],
+                                                enum: ['GET'],
```
Comment (FranekMarciniak) [inline runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/Actions/api.actions.ts]: Add delete here

**Review by FranekMarciniak — CHANGES_REQUESTED** [review]

File: runbotics/runbotics-desktop/src/config/BackgroundPageApiRequestHandler.ts [hunk 4]
```diff
@@ -56,21 +56,34 @@ export class BackgroundPageApiRequestHandler extends StatelessActionHandler impl
...
+                case 'POST':
                     response = await Axios.post(compiled, JSON.parse(input.body), { headers: input.headers });
                     break;
                 default:
                     response = await Axios.get(compiled, { headers: input.headers });
                     break;
             }
         } catch (e) {
-            throw e;
+            this.logger.log(e);
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/config/BackgroundPageApiRequestHandler.ts:82]: I dont think that we always get the response we always should throw an error. This if statement should be deleted imo

**Review by preludium — CHANGES_REQUESTED** [review]

File: runbotics-orchestrator/build.gradle [hunk 5]
```diff
@@ -1,4 +1,4 @@
-buildscript {
+    buildscript {
```
Comment (preludium) [inline runbotics-orchestrator/build.gradle]: redundant space

**Review by preludium — DISMISSED** [review]

**Review by preludium — APPROVED** [review]

**Review by FranekMarciniak — APPROVED** [review]

**Pull request merged**